File: src/test/java/net/vidageek/games/regex/task/Faileds.java
package net.vidageek.games.regex.task;

import java.util.ArrayList;
import java.util.List;

import net.vidageek.games.task.JudgedTask;

public class Faileds implements JudgedTask {

	private List<Failed> faileds = new ArrayList<Failed>(); 
	
	public boolean ok() {
		return this.faileds.isEmpty();
	}

	public String getReason() {
		
		return null;
	}

	public void addOnlyJudgedFailed(JudgedTask judgedTask) {
		if (judgedTask instanceof Failed)
			this.faileds.add((Failed)judgedTask);
	}

}

File: src/main/scala/vggames/shared/task/status/Faileds.scala
package vggames.shared.task.status

import vggames.shared.task.JudgedTask

class Faileds extends JudgedTask {

  private var faileds = List[Failed]()

  def ok : Boolean = faileds.isEmpty

  def reason : String = faileds.mkString("<br />")

  def addOnlyJudgedFailed(judgedTask : JudgedTask) {
    judgedTask match {
      case task : Failed => faileds = faileds :+ task
      case _ =>
    }
  }

  def addAll(faileds : Faileds) : Unit = faileds.foreach(addOnlyJudgedFailed)

  def judgment() : JudgedTask = if (ok) new Ok() else new Failed(this)

  private def foreach = faileds.foreach _

  def mkString = faileds.mkString(_ : String)
}
